codalab/rest/bundle.py

```diff
@@ -77,3 +86,81 @@ def get_blob(uuid, path=''):
     response.set_header('Content-Encoding', encoding or 'identity')
 
     return fileobj
+
+
+@put('/bundle/<uuid:re:%s>/contents/blob/' % spec_util.UUID_STR,
+     apply=AuthenticatedPlugin())
+def put_blob(uuid):
+    try:
+        check_bundles_have_all_permission(local.model, request.user, [uuid])
+    except PermissionError as e:
+        abort(httplib.FORBIDDEN, e.message)
+
+    bundle = safe_get_bundle(uuid)
+
+    # If this bundle already has data, remove it.
+    bundle_path = os.path.realpath(local.bundle_store.get_bundle_location(uuid))
+    if os.path.exists(bundle_path):
+        local.bundle_store.cleanup(uuid, dry_run=False)
+        bundle_update = {
+            'data_hash': None,
+        }
+        local.model.update_bundle(bundle, bundle_update)
+
+    # Figure out what kind of upload it is.
+    tar_archive = False
+    single_file_archive = False
+    if 'Content-Disposition' in request.headers:
```

Comment:
Do we always want to unpack tar.gz and gz?  What if we have a bundle whose contents is a tar.gz?  Should the query params specify?

In general, what do we expect filename to be set to?  Something that's related to the bundle name?  Please document.